receiver/kafkareceiver/config.go

```diff
@@ -74,6 +79,9 @@ type Config struct {
 
 	// Extract headers from kafka records
 	HeaderExtraction HeaderExtraction `mapstructure:"header_extraction"`
+
+	// AVRO encoder config when "encoding: avro"
+	Avro Avro `mapstructure:"avro"`
```

Comment:
imo the avro encoding should have the config defined on itself, and we should refer to the encoding extension ID. See https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/fileexporter/config.go#L45 for inspiration

Reply:
is someone already working on supporting the encoding extensions in the kafka receiver or exporter? the changes introduced by this pr are more or less temporary and can be skipped if it could be used soon. but i would prefer not mix both intents.

Reply:
I was hoping to count on you, as you seem like you have a good handle on things.

Reply:
sure, i'll have a look.